cron_scripts/inactive_idle_accounts.pl

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use lib 'lib';
+
+use Getopt::Long;
+use ZCS::API;
+use Data::Dumper;
+use JSON;
+use File::Slurp;
+use POSIX qw(strftime);
+
+my $config_file = 'config/config.json';
+
+my %opts = (
+    last_login_days => 97,
+    new_status      => 'locked'
+);
+
+my $res =
+  GetOptions( \%opts, "help", "debug", "last_login_days=i", "new_status=s" );
+
+if ( $opts{help} ) {
+    print_usage();
+    exit 0;
+}
+elsif ( !$res ) {
+    print_usage();
+    exit 1;
+}
+
+my $config = JSON->new->utf8->relaxed->decode( scalar read_file $config_file);
+die "MAILSTOREHOST is not defined in config file"
+  if ( !defined( $config->{MAILSTOREHOST} ) || $config->{MAILSTOREHOST} eq "" );
+die "MAILSTOREUSER is not defined in config file"
+  if ( !defined( $config->{MAILSTOREUSER} ) || $config->{MAILSTOREUSER} eq "" );
+die "MAILSTOREPWD is not defined in config file"
+  if ( !defined( $config->{MAILSTOREPWD} ) || $config->{MAILSTOREPWD} eq "" );
+
+my $new_status      = $opts{new_status};
+my $last_login_days = $opts{last_login_days};
+my $last_login_timestamp =
+  strftime( '%Y%m%d000000',
+    localtime( time() - $last_login_days * ( 24 * 60 * 60 ) ) )
```

Comment:
nit: perhaps not a big deal, but you're using localtime() but setting a timestamp in Z (zulu/utc)... that can lead to some slightly off timestamps (~+/- 12hrs depending)

Reply:
Do you need a sanity check to ensure the number of days isn't something crazy -1, 0, "", etc.?